pandas/core/series.py

```diff
@@ -301,6 +301,20 @@ def _update_inplace(self, result, **kwargs):
         # we want to call the generic version and not the IndexOpsMixin
         return generic.NDFrame._update_inplace(self, result, **kwargs)
 
+    @property
+    def name(self):
+        return self._name
+
+    @name.setter
+    def name(self, value):
+        if value is not None:
+            if com.is_hashable(value):
```

Comment:
That's why the pickling tests are important. I *think* this should work (as old pickles have a ``.name`` instance variable); we are now converting it when set to a ``._name``, but should be transparent to the user.